### PR TITLE
Explained how to run tests in multiple kernel apps

### LINKED ISSUE #6279: WebTestCase with multiple Microkernels
https://github.com/symfony/symfony/issues/17778#issuecomment-184882079

document how to use WebTestCase with multiple Microkernels

for reference if someone else needs it - add all Kernels to the ClassMap:

``` json
"autoload":{
    "psr-0":{
        "": "src/"
    },
    "classmap": ["app/ApiKernel.php","app/AppKernel.php","app/AppCache.php"]
}
```

now simply override getKernelClass

``` php
use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;

class MyTestCase extends WebTestCase
{
    /**
     * @return mixed
     */
    protected static function getKernelClass()
    {
        return 'ApiKernel';
    }
}
```

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #6279.

### CODE REVIEW

File: configuration/multiple_kernels.rst [hunk 1]
```diff
@@ -178,6 +178,31 @@ In order to solve this issue, add the following configuration to your kernel:
...
             "%kernel.root_dir%/../app/Resources/views": ~
 
+Running Tests Using a Different Kernel
+--------------------------------------
+
+In Symfony applications, functional tests extend by default from the
+:class:`Symfony\\Bundle\\FrameworkBundle\\Test\\WebTestCase` class. Inside that
+class, a method called ``getKernelClass()`` tries to find the class of the kernel
+to use to run the application during tests. The logic of this method does not
+support multiple kernel applications, so your tests won't use the right kernel.
```
Comment (xabbuh) [inline configuration/multiple_kernels.rst]: We need to be careful here. The approach described here does not solve the issue when you have more than one kernel. The problem then still is that the `KernelTestCase` class keeps a reference to the previously created kernel in its static `$kernel` property. Thus, if your functional tests do not run in isolated processes a later run test for a different kernel will reuse the previously created instance which points to a different kernel.

Comment (javiereguiluz) [inline configuration/multiple_kernels.rst]: Thanks for the explanation! You are right and this is critical. What would be the best (and simplest) solution to this problem?

Comment (xabbuh) [inline configuration/multiple_kernels.rst]: It should be sufficient to always reset the static `$class` property inside the `tearDown()` method:

```php
protected function tearDown()
{
    parent::tearDown();

    static::$class = null;
}
```

The `parent::tearDown()` call is important to make sure that the kernel is shut down after the test was run.

Comment (javiereguiluz) [inline configuration/multiple_kernels.rst]: Thanks!! I've fixed the code and reused your original explanation because it's great.

**Review by xabbuh — APPROVED** [review]

**Review by xabbuh — APPROVED** [review]

**Comment by xabbuh** [comment]
Thank you Javier.

**Pull request closed without merging**